### PR TITLE
Scala: fix comment-only files printing their comment twice

### PR DESCRIPTION
**Pull request by timtebeek**
A Scala file containing nothing but whitespace and comments produced both a `J.Unknown` holding the entire source and an EOF space holding the same text, so the comment was printed twice and the file failed to parse. `ScalaASTConverter.getRemainingSource` returned `""` whenever dotty produced an empty tree, which masked the duplication for `.scala` sources — `.sbt` sources parse to the `object __SbtScript__` wrapper, so their tree is non-empty and the duplicate was emitted.

Such a file now maps to a compilation unit with an empty statement list and the comments in EOF rather than a whole-file `J.Unknown` (which `rewrite-scala/CLAUDE.md` forbids), and EOF space is built with `ScalaSpace.format` so trailing comments become real `Comment` elements. Whitespace-only and empty files, which previously printed as empty and failed to parse, round-trip too.

Added `CommentOnlyFileTest` covering `//`- and `/* */`-only files on both the derived `.sbt` path and an explicit `.scala` path, with and without a trailing newline, whitespace-only/empty files, and comments leading or trailing a real declaration; 10 of its 14 cases fail without this fix. `./gradlew :rewrite-scala:test` is green (950 tests).

- Note for openrewrite/rewrite#8568 task A3 (enabling `TypeValidation.unknown()` for Scala): comment-only files no longer hit the fallback, but 51 whole-file `J.Unknown`s remain for bare top-level expressions (`42`, `"hello"`, `(a + b) * c`) because `ScalaASTConverter` keeps only conversions that are a `Statement`; wrapping those in the existing `S.ExpressionStatement` looks like the fix, and is left out of scope here.

### CODE REVIEW

File: rewrite-scala/src/main/java/org/openrewrite/scala/ScalaParserVisitor.java [hunk 1]
```diff
@@ -100,29 +100,33 @@ public S.CompilationUnit visitCompilationUnit(ScalaParseResult parseResult) {
             }
         }
 
-        // If we didn't get any statements and have source content, create an Unknown node
-        // But skip if we already have a package declaration (to avoid duplication)
```
Comment (greg-at-moderne) [inline rewrite-scala/src/main/java/org/openrewrite/scala/ScalaParserVisitor.java:104]: Based on this line alone, I'd suggest adding a test case, where the whole source is a package declaration and then some whitespace/comments. Probably already handled, but worth checking.

**Review by greg-at-moderne — APPROVED** [review]

File: rewrite-scala/src/main/java/org/openrewrite/scala/ScalaParserVisitor.java (hunk 1 above)
Comment (timtebeek) [reply rewrite-scala/src/main/java/org/openrewrite/scala/ScalaParserVisitor.java:104]: Added `packageDeclarationFollowedByComments` (5 cases): `package foo` followed by a line comment, a block comment, two comments, a comment with no trailing newline, and trailing whitespace only.

It was not fully handled — 4 of the 5 fail against `main`. The package path skips the `J.Unknown` fallback, so the round-trip was already correct, but EOF was built with `Space.build(text, emptyList())`, which stuffs the comment into whitespace: `getEof().getComments()` came back empty. The switch to `ScalaSpace.format` in this PR makes them real `Comment` elements, so recipes can see them.

Comment (greg-at-moderne) [reply rewrite-scala/src/main/java/org/openrewrite/scala/ScalaParserVisitor.java:104]: OK. Cool. Glad that I mentioned this then.

**Review by greg-at-moderne — APPROVED** [review]
